tests/test_weather_plugin.py

```
@@ -1577,6 +1581,159 @@ async def test_handle_meshtastic_meshtastic_unavailable(self):
+    @patch("mmrelay.plugins.weather_plugin.requests.get")
+    def test_generate_forecast_appends_marine_data_at_sea(self, mock_get):
+        """generate_forecast should append marine data when marine API returns data."""
+        marine_payload = {
+            "current": {
+                "wave_height": 1.2,
+                "wave_direction": 270.0,
+                "wave_period": 7.5,
+            }
+        }
+
+        def side_effect(url, **kwargs):
```

Ruff (0.15.10)

[warning] Missing return type annotation for private function `side_effect`

(ANN202)